istioctl/cmd/kubeinject.go

```diff
@@ -43,6 +59,125 @@ const (
 	valuesConfigMapKey = "values"
 )
 
+type ExternalInjector struct {
+	client       kube.ExtendedClient
+	clientConfig *admissionregistration.WebhookClientConfig
+}
+
+func (e ExternalInjector) Inject(pod *corev1.Pod) ([]byte, error) {
+	cc := e.clientConfig
+	if cc == nil {
+		return nil, fmt.Errorf("external injector is not available")
+	}
+	var address string
+	if cc.URL != nil {
+		address = *cc.URL
+	}
+	if cc.Service != nil {
+		svc, err := e.client.CoreV1().Services(cc.Service.Namespace).Get(context.Background(), cc.Service.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		namespace, selector, err := polymorphichelpers.SelectorsForObject(svc)
+		if err != nil {
+			return nil, fmt.Errorf("cannot attach to %T: %v", svc, err)
+		}
+		sortBy := func(pods []*corev1.Pod) sort.Interface { return sort.Reverse(podutils.ActivePods(pods)) }
+		pod, _, err := polymorphichelpers.GetFirstPod(e.client.CoreV1(), namespace, selector.String(), timeout, sortBy)
+		if err != nil {
+			return nil, err
+		}
+		podPort := 15017
+		for _, v := range svc.Spec.Ports {
+			if v.Name == webhookPortname {
+				podPort = v.TargetPort.IntValue()
+			}
+		}
+		f, err := e.client.NewPortForwarder(pod.Name, pod.Namespace, "", 0, podPort)
+		if err != nil {
+			return nil, err
+		}
+		if err := f.Start(); err != nil {
+			return nil, err
+		}
+		address = fmt.Sprintf("https://%s%s", f.Address(), *cc.Service.Path)
```

Comment:
Maybe instead of reverse looking of pod, its simpler to run `kubectl get --raw /api/v1/namespaces/istio-system/services/https:istiod:https-webhook/proxy/inject` equivalent? You can do it in client-go as well

Reply:
you mean do not use port-forward?  ``` kubectl get --raw /api/v1/namespaces/istio-system/services/https:istiod:https-webhook/proxy/inject
Error from server (ServiceUnavailable): no endpoints available for service "https:istiod:https-webhook"``  `  also tried `kubectl get --raw /api/v1/namespaces/istio-system/services/istiod:https-webhook/proxy/inject`and fails with similar message.

Reply:
kubectl get --raw /api/v1/namespaces/istio-system/services/https:istiod:https-webhook/proxy/inject
Error from server (BadRequest): the server rejected our request for an unknown reason

Reply:
Thats fine, if you add `-v5` you can see its an error from istiod because you didn't provide a JSON body.

Reply:
thanks